src/HasBinaryUuid.php

```diff
@@ -73,17 +73,90 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        if (! $this->exists) {
-            return parent::toArray();
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+
+        // if (! $this->exists) {
+        // return $array;
+        // }
+
+        if (is_array($uuidAttributes)) {
```

Comment:
I'd prefer an early return here. `if (! is_array()) { return ... }`

Reply:
yes, you're right and i will do this... very good refactor suggestion